vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java

```diff
@@ -103,22 +108,30 @@ public void onMediaMessagePreviewClicked(@NonNull final RoomMediaMessage roomMed
         }
     }
 
-    //endregion
-
-    //region Private methods
-
     private void updatePreview(final RoomMediaMessage roomMediaMessage) {
         mPreviewerVideoView.pause();
         mFileNameView.setText(roomMediaMessage.getFileName(this));
         final String mimeType = roomMediaMessage.getMimeType(this);
         final Uri uri = roomMediaMessage.getUri();
+
+        RequestOptions options = new RequestOptions();
+        options.fitCenter();
         if (mimeType != null) {
-            if (mimeType.startsWith("image")) {
+            if (mimeType.startsWith("image") && !mimeType.contains("gif")) {
```

Comment:
Thanks for your code. Normally Glide is supposed to work with .gif without specifying it is one (see SplashActivity). Might be an issue with the lib.
You might refactor your code like that:
```
final RequestOptions requestOptions = new RequestOptions().fitCenter();
        final RequestBuilder<? extends Drawable> requestBuilder;
        if (mimeType.startsWith("image")) {
            if (mimeType.contains("gif")) {
                requestBuilder = Glide.with(this).asGif();
            } else {
                requestBuilder = Glide.with(this).asDrawable();
            }
            mPreviewerImageView.setVisibility(View.VISIBLE);
            mPreviewerVideoView.setVisibility(View.GONE);
            requestBuilder
                    .apply(requestOptions)
                    .load(uri)
                    .into(mPreviewerImageView);
        } else if (mimeType.startsWith("video")) {...}
```

Reply:
Have you checked that it's still working? @ganfra has made some test and it was not working.

Reply:
Just checked now and it's working well